### PR TITLE
New blog post: "install keras environment from scratch"

### PR DESCRIPTION
**Pull request by FabienLavocat**
New blog post on Installing Keras on a new OS, from installing Ubuntu to PyCharm

### CODE REVIEW

File: content/install-keras-environment-from-scratch.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
+Title: Install a Keras environment from scratch
+Date: 2016-03-09
+Category: News
```
Comment (fchollet) [inline content/install-keras-environment-from-scratch.md]: This should be "tutorials".

File: content/install-keras-environment-from-scratch.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+# Install Theano (backend for Keras)
+# The other option is to TensorFlow (from Google) but we will see that option in a future post
+sudo pip install git+git://github.com/Theano/Theano.git
+
+# Install Keras
+sudo pip install keras
+```
+
+Here you are! You've installed the minimal packages to run your first script using Keras. To make sure everything works, keep your terminal open and enter the following commands:
```
Comment (fchollet) [inline content/install-keras-environment-from-scratch.md]: This does not cover running things on GPU, which is fundamental to getting things done in deep learning (and is the part of the installation that people usually need help with). We would need to cover:
- installing the Nvidia Cuda drivers
- installing cuDNN
- installing TensorFlow, GPU version
- making sure that Keras is running on GPU (with cuDNN enabled) for both TensorFlow and Theano

Ideally we should make two sections: one about setting up Keras with the Theano backend, one about setting up Keras with the TensorFlow backend.

File: content/install-keras-environment-from-scratch.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+sudo apt-get install libatlas-dev liblapack-dev
+
+# Install numpy and scipy. Those two packages might take a while to install
+sudo pip install numpy
+sudo pip install scipy
+
+# Install git to get the latest version of the next package
+sudo apt-get install git
+
+# Install Theano (backend for Keras)
```
Comment (fchollet) [inline content/install-keras-environment-from-scratch.md]: I think we should cover the installation of TensorFlow as well (including the GPU version).

File: content/install-keras-environment-from-scratch.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+Congratulation, you're on the Ubuntu desktop ready to install Keras. You might have some updates to install. Then press the **Install Now** button.
+
+![Ubuntu - Updates](/img/setup-keras/install/12-ubuntu-updates.png)
+
+## Install Keras and its dependencies
+
+First, you must open a **Terminal** and run the script bellow, line by line to install all the requirements of Keras and then Keras itself.
+
+```python
```
Comment (fchollet) [inline content/install-keras-environment-from-scratch.md]: This is a command line, the language should be specified as `sh`. Also applies to the code below.

File: content/install-keras-environment-from-scratch.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
+Title: Install a Keras environment from scratch
```
Comment (fchollet) [inline content/install-keras-environment-from-scratch.md]: I would call it "Set up a Keras Deep Learning environment from scratch". At least "set up": you don't install an environment.

**Comment by fchollet** [comment]
Looks great, but I think we need basically the following two sections (each with 2 sub-sections):
- setting up Keras on a workstation with a GPU
  - install Cuda
  - install cuDNN (v3)
  - Theano version:
    - GPU-specific Theano installation instructions 
    - make sure Keras is running on Theano with cuDNN/on GPU
  - TensorFlow version:
    - install the GPU-specific TF version (which is harder to install)
    - make sure Keras is running with cuDNN/on GPU
- setting up Keras to run only on CPU:
  - Theano version (this one bullet point is really what the tutorial is currently about)
  - TF version

File: content/install-keras-environment-from-scratch.md (hunk 1 above)
Comment (FabienLavocat) [inline content/install-keras-environment-from-scratch.md]: I've changed the category

File: content/install-keras-environment-from-scratch.md (hunk 5 above)
Comment (FabienLavocat) [inline content/install-keras-environment-from-scratch.md]: I've changed the title

File: content/install-keras-environment-from-scratch.md (hunk 3 above)
Comment (FabienLavocat) [inline content/install-keras-environment-from-scratch.md]: I was thinking about doing a different post for that. This one is about setting up a very first deep learning environment for Deep Learning with Keras. This is the very minimal stuff you need to do/install to make it work.

**Comment by FabienLavocat** [comment]
I 100% agree with you @fchollet about the two new sections. As I said in a previous comment. I think we should keep this article as the very minimal to start playing with Keras.
A second article to let people install TensorFlow and use the GPU will be required, but in a second article (IMO)

**Comment by fchollet** [comment]
> I think we should keep this article as the very minimal to start playing with Keras.
> A second article to let people install TensorFlow and use the GPU will be required, but in a second article (IMO)

Fair point. But I think it would be easier to make the different sections all part of a single "manual" (which would mean releasing the two articles at the same time, and joining them either through highlighted links ("looking for the GP version? Here") or by literally putting them on the same page.

It would be a bad strategy to release info about CPU installs (which people are generally able to do without issue in 2 lines of bash) and nothing about GPU. [People don't like that](https://www.reddit.com/r/MachineLearning/comments/462p41/pros_and_cons_of_keras_vs_lasagne_for_deep/d027u0l).

PS: what do you think about using Anaconda to simplify the install? With Anaconda, which a lot of people use, the install becomes just two lines of bash. Pros/cons?

**Pull request closed without merging**